cmd/gitlab.go

```diff
@@ -438,44 +224,59 @@ func (c *Client) pollProjects(until <-chan bool) {
 
 func (c *Client) discoverWildcards() {
 	log.Infof("%d wildcard(s) configured for polling", len(cfg.Wildcards))
-	if err := c.findProjectsFromWildcards(); err != nil {
-		log.Errorf("%v", err)
+	for _, w := range cfg.Wildcards {
+		foundProjects, err := c.listProjects(&w)
+		if err != nil {
+			log.Errorf("could not list projects for wildcard %v: %v", w, err)
+			continue
+		}
+		for _, p := range foundProjects {
+			if !projectExists(p, cfg.Projects) {
+				log.Infof("Found new project: %s", p.Name)
+				cfg.Projects = append(cfg.Projects, p)
+			}
+		}
 	}
 }
 
-func (c *Client) pollWithWorkersUntil(stop <-chan struct{}) {
-	log.Infof("%d project(s) configured for polling", len(cfg.Projects))
-	pollErrors := c.pollProjectsWith(cfg.MaximumProjectsPollingWorkers, c.pollProject, stop, cfg.Projects...)
-	for err := range pollErrors {
+func (c *Client) pollPipelinesOnInit() {
+	log.Debug("Polling latest pipelines to get data out of them")
+	for _, p := range cfg.Projects {
+		log.Debugf("On init: reading project %s", p.Name)
+		gitlabProject, err := c.getProject(p.Name)
 		if err != nil {
-			log.Errorf("%v", err)
+			log.Errorf("could not get GitLab project with name %s: %v", p.Name, err)
+			continue
+		}
+		pipeRefs, err := c.refsFromPipelines(gitlabProject, cfg.OnInitFetchRefsFromPipelinesDepthLimit)
```

Comment:
it caused me a braincycle to understand that pipe meant pipeline, perhaps we should use the full word ? 🤷

Reply:
legit 👍